addons/binding/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/internal/manager/DeviceStructureManager.java

```diff
@@ -0,0 +1,288 @@
+     *
+     * @throws IOException
+     * @throws ApiException
+     */
+    private void refreshDevices() throws IOException, ApiException {
+        List<Device> devices = client.getFullDevices();
+        for (Device d : devices) {
+            handleRefreshedDevice(d);
+        }
+    }
+
+    /**
+     * Refreshs the {@link Device} with the given id and stores it in the {@link DeviceStructureManager}.
+     *
+     * @param deviceId
+     * @throws IOException
+     * @throws ApiException
+     */
+    public void refreshDevice(String deviceId) throws IOException, ApiException {
+        Device d = client.getFullDeviceById(deviceId);
+        handleRefreshedDevice(d);
+    }
+
+    /**
+     * Stores the newly refreshed {@link Device} in the {@link DeviceStructureManager} structure and logs the
+     * {@link Device}s details and state, if the debug logging is enabled.
+     *
+     * @param d the {@link Device}
+     */
+    private void handleRefreshedDevice(Device d) {
+        if (InnogyBindingConstants.SUPPORTED_DEVICES.contains(d.getType())) {
+            addDeviceToStructure(d);
+        } else {
+            logger.debug("Device {}:'{}' by {} ({}) ignored - UNSUPPORTED.", d.getType(), d.getName(),
+                    d.getManufacturer(), d.getId());
+            logger.debug("====================================");
+            return;
+        }
+
+        if (d.isController()) {
+            bridgeDeviceId = d.getId();
+        }
+
+        if (logger.isDebugEnabled()) {
+            try {
+                logger.debug("Device {}:'{}' by {} ({}) loaded.", d.getType(), d.getName(), d.getManufacturer(),
+                        d.getId());
+                for (Capability c : d.getCapabilityMap().values()) {
+                    logger.debug("> CAP: {}/{} ({})", c.getType(), c.getName(), c.getId());
+                    if (d.isRadioDevice() && !d.isReachable()) {
+                        logger.debug(">> CAP-State: unknown (device NOT REACHABLE).");
+                    } else {
+                        if (c.getCapabilityState() != null) {
+                            for (Property p : c.getCapabilityState().getStateMap().values()) {
+                                logger.debug(">> CAP-State: {} -> {} ({})", p.getName(), p.getValue(),
+                                        p.getLastchanged());
+                            }
+                        } else {
+                            logger.debug(">> CAP-State: unknown (NULL)");
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("EX: ", e);
+            }
+            logger.debug("====================================");
+        }
+    }
+
+    /**
+     * Writes an exemplary openHAB configuration for the found {@link Device}s to the logfile
+     * which then can be copied to the config files easily for initial use of the things.
+     */
+    private void writeExampleConfigToLog() {
```

Comment:
I am not at all a fan of that feature.
It would be much nicer if people simply use the approval in the Paper UI to get their things defined.
And for people that want to switch from database to textual configuration, I would rather want to see a Thing-to-DSL exporter feature, which is independent of a specific binding, but which could write a .things-file for any give bridge with its things in the system. Would you think that this is feasible or is there any good reason to have that coded specifically for innogy (and possibly for 200 other bindings) as done here?

Reply:
If you'd be interesting to work on such a console command that could be used for any binding then, it would be cool! Feel free to create an issue or directly a PR at ESH to follow up on this.